packages/lesswrong/components/comments/RecentCommentsItem.jsx

```diff
@@ -24,19 +14,26 @@ class RecentCommentsItem extends getRawComponent('CommentsItem') {
       showReply: false,
       showEdit: false,
       expanded: false,
+      showParent: false
     };
   }
 
+  toggleShowParent = () => {
+    this.setState({showParent:!this.state.showParent})
+  }
+
   renderRecentComment() {
-    let content = this.props.comment.content;
-    const htmlBody = {__html: this.props.comment.htmlBody};
-    const plaintext = this.props.comment.body;
-    if (!this.state.expanded && plaintext && plaintext.length > 300) {
+    const comment = this.props.comment || this.props.document
```

Comment:
Hmm, I feel like we want to have a uniform API here. Any reason why we need to support both here?

Reply:
So, we need a way to render comments with parentCommentId (so that they can work recursively without having to query the entire thread). 

Previously the RecentCommentItem was used for lists that were populated fully by the list api call. We could make them two different components, or we could rework the list thing to instead pass in Ids that RecentCommentItem uses.